contracts/strategy/StrategyStrictBase.sol

```diff
@@ -0,0 +1,257 @@
+
+    balance = _checkWithdrawImpact(
+      _asset,
+      balance,
+      investedAssetsUSD,
+      assetPrice
+    );
+
+    {
+      // we cannot withdraw more than the base amount value
+      // if any additional amount exist on the balance (i.e. airdrops)
+      // it should be processed by hardwork at first (split on compound/forwarder)
+      uint baseAmount = baseAmounts[_asset];
+      if (balance > baseAmount) {
+        balance = baseAmount;
+      }
+    }
+
+    if (balance != 0) {
+      _decreaseBaseAmount(_asset, balance);
+      IERC20(_asset).safeTransfer(_vault, balance);
+    }
+    emit WithdrawAllToVault(balance);
+  }
+
+  /// @dev Withdraws some assets to the vault
+  function withdrawToVault(uint amount) external override {
+    address _vault = vault;
+    address _asset = asset; // gas saving
+    require(msg.sender == _vault, DENIED);
+    uint balance = IERC20(_asset).balanceOf(address(this));
+    if (amount > balance) {
+      (uint investedAssetsUSD, uint assetPrice) = _withdrawFromPool(amount - balance);
+      balance = _checkWithdrawImpact(
+        _asset,
+        balance,
+        investedAssetsUSD,
+        assetPrice
+      );
+    }
+
+    uint amountAdjusted = Math.min(amount, balance);
+    if (amountAdjusted != 0) {
+      _decreaseBaseAmount(_asset, amountAdjusted);
+      IERC20(_asset).safeTransfer(_vault, amountAdjusted);
+    }
+    emit WithdrawToVault(amount, amountAdjusted, balance);
+  }
+
+
+  // *************************************************************
+  //                  baseAmounts modifications
+  // *************************************************************
+
+  /// @notice Decrease {baseAmounts} of the {asset} on {amount_}
+  ///         The {amount_} can be greater then total base amount value because it can includes rewards.
+  ///         We assume here, that base amounts are spent first, then rewards and any other profit-amounts
+  function _decreaseBaseAmount(address asset_, uint amount_) internal {
+    uint baseAmount = baseAmounts[asset_];
+    require(baseAmount >= amount_, WRONG_AMOUNT);
+    baseAmounts[asset_] = baseAmount - amount_;
+    emit UpdateBaseAmounts(asset_, -int(baseAmount));
+  }
+
+  /// @notice Increase {baseAmounts} of the {asset} on {amount_}, ensure that current {assetBalance_} >= {amount_}
+  /// @param assetBalance_ Current balance of the {asset} to check if {amount_} > the balance. Pass 0 to skip the check
+  function _increaseBaseAmount(address asset_, uint amount_, uint assetBalance_) internal {
+    baseAmounts[asset_] += amount_;
+    emit UpdateBaseAmounts(asset_, int(amount_));
+    require(assetBalance_ >= amount_, WRONG_AMOUNT);
+  }
+
+  // *************************************************************
+  //                       HELPERS
+  // *************************************************************
+
+  // todo: not sure if we need this function since priceChangeTolerance is always 0 for strict strategies.
```

Comment:
@belbix Please take a look. I am not sure if this function is needed since there are no vault fees. But we could have some discrepancies from v1 vaults. We can have some hardcoded value instead of uint priceChangeTolerance = 0;

Reply:
since it is strict, let's make sure that the share price NEVER goes down